src/main/java/uk/nhs/hee/tis/trainee/sync/facade/ProgrammeMembershipEnricherFacade.java

```diff
@@ -187,558 +71,96 @@ public void enrich(Programme programme) {
+    List<AggregateCurriculumMembershipDto> aggregatedCurriculumMemberships =
+        buildCurriculumMemberships(programmeMembership);
+    Programme programme = getProgramme(programmeMembership);
+
+    if (!aggregatedCurriculumMemberships.isEmpty() && programme != null) {
+      // TODO: validate the aggregated data to ensure we have a "complete" PM?
+      AggregateProgrammeMembershipDto aggregateProgrammeMembership =
+          aggregateMapper.toAggregateProgrammeMembershipDto(programmeMembership, programme,
+              aggregatedCurriculumMemberships);
+      syncProgrammeMembership(aggregateProgrammeMembership);
+    }
   }
 
   /**
-   * Sync an enriched programmeMembership with the programmeMembership. Optionally enrich programme
-   * or curriculum details Optionally completely resync all programme memberships for the person
+   * Get the aggregated curriculum membership data for the given programme membership, any missing
+   * data will be requested.
    *
-   * @param programmeMembership                  The programmeMembership to enrich.
-   * @param doProgrammeEnrich                    Enrich programme details
-   * @param doCurriculumEnrich                   Enrich curriculum details
-   * @param doRebuildPersonsProgrammeMemberships Rebuild all programme memberships for person
+   * @param programmeMembership The programme membership to get the curriculum memberships for.
+   * @return The list of aggregated curriculum membership data, or an empty list if not all data was
+   *     available.
    */
-  private void enrich(Record programmeMembership,
-      boolean doProgrammeEnrich,
-      boolean doCurriculumEnrich,
-      boolean doRebuildPersonsProgrammeMemberships) {
-    boolean doSync = true;
-
-    if (doProgrammeEnrich) {
-      String programmeId = getProgrammeId(programmeMembership);
+  private List<AggregateCurriculumMembershipDto> buildCurriculumMemberships(
+      ProgrammeMembership programmeMembership) {
+    List<AggregateCurriculumMembershipDto> aggregatedCurriculumMemberships = new ArrayList<>();
+    boolean dataRequested = false;
 
-      if (programmeId != null) {
-        Optional<Programme> optionalProgramme = programmeSyncService.findById(programmeId);
+    String programmeMembershipUuid = programmeMembership.getUuid().toString();
+    Set<CurriculumMembership> curriculumMemberships =
+        curriculumMembershipService.findByProgrammeMembershipUuid(programmeMembershipUuid);
 
-        if (optionalProgramme.isPresent()) {
-          doSync = enrich(programmeMembership, optionalProgramme.get());
-        } else {
-          programmeSyncService.request(programmeId);
-          doSync = false;
-        }
-      }
-    }
-
-    if (doCurriculumEnrich) {
-      String curriculumId = getCurriculumId(programmeMembership);
-
-      if (curriculumId != null) {
+    if (curriculumMemberships.isEmpty()) {
+      curriculumMembershipService.requestForProgrammeMembership(programmeMembershipUuid);
```

Comment:
I realise it makes no effective difference in what is returned, but for clarity might it be worthwhile to include a `dataRequested = true` statement here?

Reply:
I didn't like the idea of adding a line of code that didn't really do anything, but I figured I could attach some logging to it and get both clarity and value.

```
if (curriculumMemberships.isEmpty()) {
  curriculumMembershipService.requestForProgrammeMembership(programmeMembershipUuid);
  dataRequested = true;
} else {
  ...
}

if (dataRequested) {
  log.info("blah blah skipped due to missing data.");
}

return dataRequested ? List.of() : aggregatedCurriculumMemberships;
```